wtf-hir/src/compiler.rs

```diff
@@ -1093,10 +1098,68 @@ impl HirCompiler {
             // TODO: put in more auto-conversions, e.g. casting from non-optional to optional should insert an explicit call to "some"
             (Type::Option(inner), Type::None) => Expression::none(*inner.clone()),
             (Type::Option(_), _) => Expression::some(actual),
-            (a, b) => {
+            // Record casting
```

Comment:
This now is a lot of duplicated code that could be compressed by using Type::Record(_) | Type::Name(_) in the pattern